### PR TITLE
Input modules implementation:

### PR DESCRIPTION
**Pull request by zemtsov-artem**
1.Keyboard module + unit tests
2.Mouse module + unit tests.
Also travis yml file updated to support OS tests.

### CODE REVIEW

File: pywinauto/macos/keyboard_helper.py [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+                    flag=None
+                elif(c in selected_keyboard.key_code_map):
+                    key_code=selected_keyboard.key_code_map[c]
+                    keys.append(KeyAction(key_code,flag=flag))
+                    flag=None
+
+    return keys
+
+
+def SendKeys(keys,
```
Comment (vasily-v-ryabov) [inline pywinauto/macos/keyboard_helper.py]: It's worth using `send_keys` name only.

File: pywinauto/macos/macos_functions.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Mac branch of the keyboard module
```
Comment (vasily-v-ryabov) [inline pywinauto/macos/macos_functions.py]: Irrelevant comment.

File: pywinauto/macos/macos_functions.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        print (iter)
+
+def check_is_process_trusted():
+    if not (AXIsProcessTrusted()):
+        if (is_debug):
+            print("Process is not Trusted")
+        exit(-1)
+
+def from_py_objc_to_apple_attribute(pyObjcName):
+    return unicode(pyObjcName)
```
Comment (vasily-v-ryabov) [inline pywinauto/macos/macos_functions.py]: Should be `six.text_type` instead of `unicode`.

File: pywinauto/macos/macos_functions.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+def print_list_of_atrributes(ax_element):
+    for iter in dir(ax_element):
+        print (iter)
+
+def check_is_process_trusted():
+    if not (AXIsProcessTrusted()):
+        if (is_debug):
+            print("Process is not Trusted")
+        exit(-1)
```
Comment (vasily-v-ryabov) [inline pywinauto/macos/macos_functions.py:96]: Exception is always better than silent exit.

File: pywinauto/mouse.py [hunk 5]
```diff
@@ -175,6 +182,47 @@ def _perform_click_input(
         if ('alt' in keyboard_keys) and key_up:
             keyboard.VirtualKeyAction(keyboard.VK_MENU, down=False).run()
 
+elif sys.platform == 'darwin':
+    def _perform_click_input(button='left', coords=(0, 0),
+                             button_down=True, button_up=True, double=False,
+                             wheel_dist=0, pressed="", key_down=True, key_up=True):
+        events = []
+        pause_value = 0.5
```
Comment (vasily-v-ryabov) [inline pywinauto/mouse.py:202]: Can it be default value from module `timings`?

File: pywinauto/unittests/test_keyboard.py [hunk 6]
```diff
@@ -46,6 +46,12 @@
     from pywinauto.keyboard import KeyAction, VirtualKeyAction, PauseAction
     from pywinauto.sysinfo import is_x64_Python, is_x64_OS
     from pywinauto.application import Application
+elif sys.platform == 'darwin':
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    sys.path.append(parent_dir)
+    sys.path.append(parent_dir + '/macos')
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_keyboard.py]: `os.path.join`

File: pywinauto/unittests/test_keyboard.py [hunk 7]
```diff
@@ -46,6 +46,12 @@
     from pywinauto.keyboard import KeyAction, VirtualKeyAction, PauseAction
     from pywinauto.sysinfo import is_x64_Python, is_x64_OS
     from pywinauto.application import Application
+elif sys.platform == 'darwin':
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    sys.path.append(parent_dir)
+    sys.path.append(parent_dir + '/macos')
+    import macos_functions
+    from keyboard_helper import SendKeys,KeyAction,KeySequenceError
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_keyboard.py]: Spaces after comma

File: pywinauto/unittests/test_keyboard.py [hunk 8]
```diff
@@ -113,6 +126,11 @@ def receive_text(self):
         received = ' '
         if sys.platform == 'win32':
             received = self.ctrl.TextBlock()
+        elif sys.platform == 'darwin':
+            # Clear clipboard
+            os.system("pbcopy < /dev/null")
+            SendKeys('{cmd}a{cmd}x',pause=0.2)
+            received = macos_functions.read_from_clipboard()
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_keyboard.py]: `read_from_clipboard` can be moved to macOS branch in `clipboard.py` and aligned with naming in this module.

File: pywinauto/unittests/test_mouse.py [hunk 9]
```diff
@@ -12,6 +12,14 @@
     from pywinauto.keyboard import SendKeys
     from pywinauto import mouse
     from pywinauto.timings import Timings
+elif sys.platform == 'darwin':
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    sys.path.append(parent_dir)
+    import mouse
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_mouse.py:18]: `from .. import mouse` not working? Using `sys.path.append` everywhere is not a good practice.

File: pywinauto/unittests/test_mouse.py [hunk 10]
```diff
@@ -12,6 +12,14 @@
...
+    sys.path.append(parent_dir + '/macos')
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_mouse.py:19]: ->`os.path.join`

File: pywinauto/unittests/test_mouse.py [hunk 11]
```diff
@@ -45,19 +53,31 @@ def setUp(self):
             self.app = Application()
             self.app.start(_test_app())
             self.dlg = self.app.mousebuttons
+        elif sys.platform == 'darwin':
+            macos_functions.launch_application("mousebuttons")
+            self.app = macos_functions.get_instance_of_app("mousebuttons")
+            os.system("ps -ax | grep mousebuttons")
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_mouse.py:59]: -> `output = subprocess.check_output(...)`

File: pywinauto/macos/keyboard_helper.py (hunk 1 above)
Comment (zemtsov-artem) [reply pywinauto/macos/keyboard_helper.py]: @vasily-v-ryabov should I also rename this method in linux/base_wrapper ?

Comment (vasily-v-ryabov) [reply pywinauto/macos/keyboard_helper.py]: Yes, it makes sense.

**Pull request closed without merging**